### PR TITLE
fix(wallet): account derivations

### PR DESCRIPTION
**Pull request by plondon**
## Description (optional)
Some users have reported incomplete account derivations or missing derivations list. This PR finds those accounts on login and fixes them as long as user does not have a 2nd password.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

TODO
- [x] test multiple accounts w/ bad derivations
- [x] check that the derivation is correct for account

### CODE REVIEW

File: packages/blockchain-wallet-v4/src/redux/wallet/sagas.ts [hunk 1]
```diff
@@ -310,6 +312,40 @@ export default ({ api, networks }) => {
...
   }
 
+  const getAccountsWithIncompleteDerivations = function* () {
+    const isEncrypted = yield select(S.isSecondPasswordOn)
+    if (isEncrypted) return []
+
+    const wallet = yield select(S.getWallet)
+    const accounts = Wallet.selectHDAccounts(wallet)
+    const accountsWithMissingDerivations = accounts
+      .filter((acct) => acct.derivations.size < DERIVATION_LIST.length)
```
Comment (plondon) [inline packages/blockchain-wallet-v4/src/redux/wallet/sagas.ts:322]: Check explicitly 0

**Review by mpfluger — APPROVED** [review]

**Review by mpfluger — APPROVED** [review]

**Review by schnogz — APPROVED** [review]

File: packages/blockchain-wallet-v4/src/redux/wallet/reducers.ts [hunk 2]
```diff
@@ -64,6 +64,10 @@ export const wrapperReducer = (state = WRAPPER_INITIAL_STATE, action) => {
       const { accountIdx, archived } = action.payload
       return over(Wrapper.wallet, Wallet.setAccountArchived(accountIdx, archived), state)
     }
+    case AT.SET_ACCOUNT_DERIVATIONS: {
+      const { accountIdx, derivations } = action.payload
```
Comment (paulo-bc) [inline packages/blockchain-wallet-v4/src/redux/wallet/reducers.ts:68]: @schnogz We are missing setting `default_derivation` here.

File: packages/blockchain-wallet-v4/src/redux/wallet/sagas.ts [hunk 3]
```diff
@@ -310,6 +312,40 @@ export default ({ api, networks }) => {
...
+      if (isEncrypted) return
+
+      const getSeedHex = yield select(S.getSeedHex, null)
+      const seedHex = yield call(() => taskToPromise(getSeedHex))
+
+      // eslint-disable-next-line no-restricted-syntax
+      for (const acct of accounts) {
+        const accountIdx = acct.index
+        const derivations = HDWallet.generateDerivations(seedHex, accountIdx)
+        yield put(A.wallet.setAccountDerivations(accountIdx, DerivationList.fromJS(derivations)))
```
Comment (paulo-bc) [inline packages/blockchain-wallet-v4/src/redux/wallet/sagas.ts:342]: @schnogz We are missing setting default_derivation here.

File: packages/blockchain-wallet-v4/src/redux/wallet/reducers.ts (hunk 2 above)
Comment (paulo-bc) [reply packages/blockchain-wallet-v4/src/redux/wallet/reducers.ts:68]: @plondon

**Pull request merged**